draft-ietf-quic-tls.md

```diff
@@ -1168,89 +1168,226 @@ anticipation of receiving a ClientHello.
 
 # Key Update
 
-Once the handshake is confirmed, it is possible to update the keys. The
-KEY_PHASE bit in the short header is used to indicate whether key updates
-have occurred. The KEY_PHASE bit is initially set to 0 and then inverted
-with each key update.
+Once the handshake is confirmed (see {{handshake-confirmed}}), it is possible to
+update the keys used to protect packets.
 
-The KEY_PHASE bit allows a recipient to detect a change in keying material
-without necessarily needing to receive the first packet that triggered the
-change.  An endpoint that notices a changed KEY_PHASE bit can update keys and
-decrypt the packet that contains the changed bit.
+The Key Phase bit indicates which packet protection keys are used to protect the
+packet.  The Key Phase bit is initially set to 0 for the first set of 1-RTT
+packets and toggled to signal each subsequent key update.
```

Comment:
I think that this sentence, along with the sentence stating that "an endpoint MUST NOT initiate a key update prior to having received an acknowledgment for a packet that it sent protected with keys from the current key phase" is introducing a design change.

At the moment, we allow the first key update to be executed when the handshake is confirmed, while prohibiting subsequent key updates happening before receiving an ACK for the preceding key update. That has meant that if an endpoint were to send its first 1-RTT packet after the handshake has been confirmed could send the first 1-RTT packet with the Key Phase bit set to 1. The new text prohibits that.

Reply:
I don't follow this comment.  Handshake confirmation requires ACK of a packet that was sent after handshake completion.  I'll try to remove that dependency, but it is currently functionally identical.

Reply:
Yeah that's true, but it's like filling a land mine that would explode once we change the definition of handshake complete.